crates/optimism/rpc/src/sequencer.rs

```diff
@@ -19,22 +49,41 @@ pub struct SequencerClient {
 
 impl SequencerClient {
     /// Creates a new [`SequencerClient`].
-    pub fn new(sequencer_endpoint: impl Into<String>) -> Self {
-        let client = reqwest::Client::builder().use_rustls_tls().build().unwrap();
-        Self::with_client(sequencer_endpoint, client)
+    pub async fn new(sequencer_endpoint: impl Into<String>) -> Result<Self> {
+        let client = reqwest::Client::builder().use_rustls_tls().build()?;
+        Self::with_client(sequencer_endpoint, client).await
     }
```

Comment:
hmm, this creates a new request client but this can be discarded if the endpoint is a ws URL, so we can split the constructor functions a bit so that we have dedicated fns for HTTP and ws and one that checks the URL and calls the appropriate one